texlive build broken on master due to icu upgrade

When building texlive:

    nix-build '<nixpkgs>' -A texlive

I get the following errors when xetex includes ICU:

```
In file included from /nix/store/x88k2myciki60yafz4bxrdsf42mw8wrf-icu4c-59.1-dev/include/unicode/utext.h:145:
/nix/store/x88k2myciki60yafz4bxrdsf42mw8wrf-icu4c-59.1-dev/include/unicode/unistr.h:3025:7: error: delegating constructors are permitted only in C++11
      UnicodeString(ConstChar16Ptr(text)) {}
      ^~~~~~~~~~~~~
/nix/store/x88k2myciki60yafz4bxrdsf42mw8wrf-icu4c-59.1-dev/include/unicode/unistr.h:3087:7: error: delegating constructors are permitted only in C++11
      UnicodeString(ConstChar16Ptr(text), length) {}
      ^~~~~~~~~~~~~
/nix/store/x88k2myciki60yafz4bxrdsf42mw8wrf-icu4c-59.1-dev/include/unicode/unistr.h:3180:7: error: delegating constructors are permitted only in C++11
      UnicodeString(Char16Ptr(buffer), buffLength, buffCapacity) {}
      ^~~~~~~~~~~~~
```

--- a/pkgs/tools/typesetting/tex/texlive/bin.nix
+++ b/pkgs/tools/typesetting/tex/texlive/bin.nix
@@ -187,6 +187,7 @@ core-big = stdenv.mkDerivation { #TODO: upmendex
   '';
 
   preBuild = "cd texk/web2c";
+  CXXFLAGS = "-std=c++11 -Wno-reserved-user-defined-literal"; # TODO: remove once texlive 2017 is out?
   enableParallelBuilding = true;
 
   # now distribute stuff into outputs, roughly as upstream TL

